eq-author/src/components/Collapsible/index.js

```diff
@@ -81,42 +81,86 @@ const HideThisButton = styled(Button)`
   }
 `;
 
-const Collapsible = ({ title, defaultOpen, className, children }) => {
+const Collapsible = ({
+  showHide = false,
+  withoutHideThis = false,
+  title,
+  defaultOpen,
+  className,
+  children,
+}) => {
   const [isOpen, setIsOpen] = useState(defaultOpen);
 
+  const renderTitle = (showHide, isOpen, title) => {
+    if (showHide && isOpen) {
+      return `Hide ${title}`;
+    } else if (showHide && !isOpen) {
+      return `Show ${title}`;
+    }
+
+    return title;
+  };
```

Comment:
Can move this function outside the component because all of the arguments are being passed in